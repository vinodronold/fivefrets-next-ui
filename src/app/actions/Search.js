import {
  TOGGLE_SEARCH,
  CATPURE_SEARCH_INPUT,
  FETCH_SEARCH_RESULT_SUCCESS,
  FETCH_SEARCH_RESULT_ERROR
} from '../constants/ActionTypes'

export const ToggleSearch = () => ({
  type: TOGGLE_SEARCH
})

export const CaptureSearchInput = input => ({
  type: CATPURE_SEARCH_INPUT,
  input
})

export const FetchSearchResultSuccess = result => ({
  type: FETCH_SEARCH_RESULT_SUCCESS,
  result
})

export const FetchSearchResultError = () => ({
  type: FETCH_SEARCH_RESULT_ERROR
})
