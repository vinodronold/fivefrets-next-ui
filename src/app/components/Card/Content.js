import React, { PureComponent } from 'react'
import { node } from 'prop-types'
export default class Content extends PureComponent {
  static propTypes = {
    children: node.isRequired
  }
  render() {
    return (
      <div>
        {this.props.children}
        <style jsx>{`
          margin: 0 1rem;
          display: block;
        `}</style>
      </div>
    )
  }
}
