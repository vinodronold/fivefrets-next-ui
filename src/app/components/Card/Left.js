import React, { PureComponent } from 'react'
import { string } from 'prop-types'
import { YTImg } from '../../utils'

export default class Left extends PureComponent {
  static propTypes = {
    id: string.isRequired,
    height: string.isRequired
  }
  render() {
    const { id, height } = this.props
    return (
      <div>
        <img src={YTImg(id)} alt={'YT'} height={height} width={'auto'} />
        <style jsx>{`
          div {
            display: flex;
            margin-right: 1rem;
            align-items: center;
            height: auto;
            width: auto;
          }
        `}</style>
      </div>
    )
  }
}
