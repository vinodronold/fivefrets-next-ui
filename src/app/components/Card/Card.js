import React, { PureComponent } from 'react'
import { string, func } from 'prop-types'
import { color } from '../../constants/theme'
import Left from './Left'
import Content from './Content'
import { Title, SubHeading1 } from '../Typography'

export default class Card extends PureComponent {
  static propTypes = {
    id: string.isRequired,
    title: string.isRequired,
    subtitle: string,
    height: string,
    onClick: func.isRequired
  }

  render() {
    const { id, title, subtitle, height = 'auto', onClick } = this.props
    return (
      <div onClick={onClick}>
        <Left id={id} height={height} />
        <Content>
          <Title>{title}</Title>
          <SubHeading1>{subtitle}</SubHeading1>
        </Content>
        <style jsx>{`
          div {
            display: flex;
            align-items: center;
            min-height: 3rem;
            margin: 1rem auto;
            background-color: rgba(0, 0, 0, 0.65);
            cursor: pointer;
            transition: all 0.5s;
            width: 80%;
          }
          div:hover {
            color: ${color.secondary(1)};
          }
        `}</style>
      </div>
    )
  }
}
