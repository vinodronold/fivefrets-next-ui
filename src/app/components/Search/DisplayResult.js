import React, { PureComponent } from 'react'
import { shape, string, bool, func } from 'prop-types'
import Router from 'next/router'
import Loader from '../Loader'
import Card from '../Card'
import { Caption } from '../Typography'

export const handleRouteChangeComplete = ToggleSearch => {
  ToggleSearch()
  Router.onRouteChangeComplete = null
}

export default class DisplayResult extends PureComponent {
  static propTypes = {
    result: shape({
      id: string.isRequired,
      title: string.isRequired,
      subtitle: string
    }),
    IsError: bool,
    ToggleSearch: func.isRequired
  }

  render() {
    const { result = [], IsError = false, ToggleSearch } = this.props
    return IsError ? (
      <Caption style={{ textAlign: 'center' }}>Oops.. Error.. Search did not complete.. Sorry..</Caption>
    ) : result.length > 0 ? (
      <div>
        {result.map(r => (
          <Card
            key={r.id}
            {...r}
            height="45rem"
            onClick={() => {
              Router.push(`/play?id=${r.id}`, `/p/${r.id}`)
              Router.onRouteChangeComplete = handleRouteChangeComplete(ToggleSearch)
            }}
          />
        ))}
        <style jsx>{`
          div {
            padding: 1rem;
            display: block;
          }
        `}</style>
      </div>
    ) : (
      <Loader />
    )
  }
}
