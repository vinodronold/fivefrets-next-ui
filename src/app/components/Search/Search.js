import React, { PureComponent } from 'react'
import { bool, string, func, array } from 'prop-types'
import { color } from '../../constants/theme'
import DisplayResult from './DisplayResult'

export default class Search extends PureComponent {
  static propTypes = {
    isSearchOpen: bool.isRequired,
    IsError: bool.isRequired,
    ToggleSearch: func.isRequired,
    CaptureSearchInput: func.isRequired,
    result: array,
    input: string
  }

  render() {
    const { isSearchOpen, input, result, IsError, ToggleSearch, CaptureSearchInput } = this.props
    return (
      <div className={`container ${isSearchOpen ? 'open' : 'closed'}`}>
        <div className={'search'}>
          <input
            type="text"
            value={input}
            placeholder={'Search'}
            onChange={e => {
              !isSearchOpen && ToggleSearch()
              CaptureSearchInput(e.target.value)
            }}
          />
        </div>
        {input && <DisplayResult result={result} IsError={IsError} ToggleSearch={ToggleSearch} />}

        <style jsx>{`
          div.container {
            padding: 4rem;
          }
          div.container.open {
            padding: 2rem;
            position: fixed;
            background-color: ${color.bg(1)};
            top: 0;
            bottom: 0;
            left: 0;
            right: 0;
            zindex: 12;
            margintop: 3rem;
          }
          div.search {
            text-align: center;
          }
          input {
            background-color: rgba(0, 0, 0, 0);
            border-style: none;
            width: 80%;
            border-bottom: 2px solid #fff;
            color: ${color.primary()};
          }
          input:focus {
            outline: none;
          }
        `}</style>
        <style jsx>{`
          input {
            text-align: ${isSearchOpen ? 'left' : 'center'};
            font-size: ${isSearchOpen ? '4rem' : '2rem'};
          }
        `}</style>
      </div>
    )
  }
}
