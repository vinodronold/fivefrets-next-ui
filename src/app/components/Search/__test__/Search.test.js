import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Search from '../Search'
import DisplayResult, { handleRouteChangeComplete } from '../DisplayResult'
import Card from '../../Card'

jest.mock('next/router')

describe('components/Search', () => {
  describe('Search', () => {
    it('should match Search Snapshot when not open', () => {
      const _search = renderer.create(
        <Search
          isSearchOpen={false}
          IsError={false}
          result={[]}
          input={''}
          ToggleSearch={jest.fn()}
          CaptureSearchInput={jest.fn()}
        />
      )
      expect(_search).toMatchSnapshot()
    })
    it('should match Search Snapshot when open', () => {
      const _search = renderer.create(
        <Search
          isSearchOpen={true}
          IsError={false}
          result={[]}
          input={''}
          ToggleSearch={jest.fn()}
          CaptureSearchInput={jest.fn()}
        />
      )
      expect(_search).toMatchSnapshot()
    })
    it('should match Search Snapshot when Error', () => {
      const _search = renderer.create(
        <Search
          isSearchOpen={true}
          IsError={true}
          result={[]}
          input={''}
          ToggleSearch={jest.fn()}
          CaptureSearchInput={jest.fn()}
        />
      )
      expect(_search).toMatchSnapshot()
    })
    it('should match Search Snapshot when Display Results', () => {
      const _search = renderer.create(
        <Search
          isSearchOpen={true}
          IsError={true}
          result={[
            { id: 1, title: 'title 1'},
            { id: 2, title: 'title 2'}
          ]}
          input={'abc'}
          ToggleSearch={jest.fn()}
          CaptureSearchInput={jest.fn()}
        />
      )
      expect(_search).toMatchSnapshot()
    })
  })

  describe('DisplayResult', () => {
    it('should match DisplayResult snapshot', () => {
      const _display_result = renderer.create(<DisplayResult ToggleSearch={jest.fn} />)
      expect(_display_result).toMatchSnapshot()
    })
    it('should match DisplayResult snapshot with results', () => {
      const _display_result = renderer.create(
        <DisplayResult
          result={[
            { id: 1, title: 'title 1'},
            { id: 2, title: 'title 2'}
          ]}
          ToggleSearch={jest.fn}
        />
      )
      expect(_display_result).toMatchSnapshot()
    })
    it('should match DisplayResult snapshot when Error', () => {
      const _display_result = renderer.create(<DisplayResult IsError={true} ToggleSearch={jest.fn} />)
      expect(_display_result).toMatchSnapshot()
    })
    it('should match snapshot after clicking Card', () => {
      
      const _ToggleSearch = jest.fn()
      const _display_result = shallow(
        <DisplayResult result={[{ id: 1, title: 'title 1'}]} ToggleSearch={_ToggleSearch} />
      )
      _display_result.find(Card).simulate('click')
      expect(_display_result).toMatchSnapshot()
      expect(_ToggleSearch).toBeCalled()
    })
    it('handleRouteChangeComplete should call ToggleSearch', () => {
      const _ToggleSearch = jest.fn()
      handleRouteChangeComplete(_ToggleSearch)
      expect(_ToggleSearch).toBeCalled()
    })
  })

  describe('onChange Functions', () => {
    it('should match the snapshot after onChange', () => {
      const _search = shallow(
        <Search
          isSearchOpen={false}
          IsError={false}
          result={[]}
          input={''}
          ToggleSearch={jest.fn()}
          CaptureSearchInput={jest.fn()}
        />
      )
      _search.find('input').simulate('change', {
        target: {
          value: 'Search Input'
        }
      })
      expect(_search).toMatchSnapshot()
    })
  })
})
