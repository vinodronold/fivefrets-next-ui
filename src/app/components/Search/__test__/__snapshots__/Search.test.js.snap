// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Search DisplayResult should match DisplayResult snapshot 1`] = `
<div
  className="jsx-3649670468 container"
>
  <div
    className="jsx-3649670468 loader"
  >
    <div
      className="jsx-791847780 jsx-2291006144 bar"
    />
    <div
      className="jsx-791847780 jsx-3500035231 bar"
    />
    <div
      className="jsx-791847780 jsx-1901401244 bar"
    />
    <div
      className="jsx-791847780 jsx-40336921 bar"
    />
    <div
      className="jsx-791847780 jsx-3370661658 bar"
    />
  </div>
  <div
    className="jsx-4247411757 jsx-3754337244"
    style={Object {}}
  >
    LOADING...
  </div>
</div>
`;

exports[`components/Search DisplayResult should match DisplayResult snapshot when Error 1`] = `
<div
  className="jsx-4247411757 jsx-3754337244"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  Oops.. Error.. Search did not complete.. Sorry..
</div>
`;

exports[`components/Search DisplayResult should match DisplayResult snapshot with results 1`] = `
<div
  className="jsx-1201711722"
>
  <div
    className="jsx-2416660906"
    onClick={[Function]}
  >
    <div
      className="jsx-4231049512"
    >
      <img
        alt="YT"
        className="jsx-4231049512"
        height="45rem"
        src="https://i.ytimg.com/vi/1/default.jpg"
        width="auto"
      />
    </div>
    <div
      className="jsx-250553575"
    >
      <div
        className="jsx-4247411757 jsx-873826105"
        style={Object {}}
      >
        title 1
      </div>
      <div
        className="jsx-4247411757 jsx-482228210"
        style={Object {}}
      >
        subtitle 1
      </div>
    </div>
  </div>
  <div
    className="jsx-2416660906"
    onClick={[Function]}
  >
    <div
      className="jsx-4231049512"
    >
      <img
        alt="YT"
        className="jsx-4231049512"
        height="45rem"
        src="https://i.ytimg.com/vi/2/default.jpg"
        width="auto"
      />
    </div>
    <div
      className="jsx-250553575"
    >
      <div
        className="jsx-4247411757 jsx-873826105"
        style={Object {}}
      >
        title 2
      </div>
      <div
        className="jsx-4247411757 jsx-482228210"
        style={Object {}}
      >
        subtitle 2
      </div>
    </div>
  </div>
</div>
`;

exports[`components/Search DisplayResult should match snapshot after clicking Card 1`] = `
<div
  className="jsx-1201711722"
>
  <Card
    height="45rem"
    id={1}
    key="1"
    onClick={[Function]}
    subtitle="subtitle 1"
    title="title 1"
  />
  <JSXStyle
    css="div.jsx-1201711722{padding:1rem;display:block;}"
    styleId="1201711722"
  />
</div>
`;

exports[`components/Search Search should match Search Snapshot when Display Results 1`] = `
<div
  className="jsx-966561052 jsx-248342299 container open"
>
  <div
    className="jsx-966561052 jsx-248342299 search"
  >
    <input
      className="jsx-966561052 jsx-248342299"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value="abc"
    />
  </div>
  <div
    className="jsx-4247411757 jsx-3754337244"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Oops.. Error.. Search did not complete.. Sorry..
  </div>
</div>
`;

exports[`components/Search Search should match Search Snapshot when Error 1`] = `
<div
  className="jsx-966561052 jsx-248342299 container open"
>
  <div
    className="jsx-966561052 jsx-248342299 search"
  >
    <input
      className="jsx-966561052 jsx-248342299"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  
</div>
`;

exports[`components/Search Search should match Search Snapshot when not open 1`] = `
<div
  className="jsx-966561052 jsx-369727501 container closed"
>
  <div
    className="jsx-966561052 jsx-369727501 search"
  >
    <input
      className="jsx-966561052 jsx-369727501"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  
</div>
`;

exports[`components/Search Search should match Search Snapshot when open 1`] = `
<div
  className="jsx-966561052 jsx-248342299 container open"
>
  <div
    className="jsx-966561052 jsx-248342299 search"
  >
    <input
      className="jsx-966561052 jsx-248342299"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  
</div>
`;

exports[`components/Search onChange Functions should match the snapshot after onChange 1`] = `
<div
  className="jsx-966561052 jsx-369727501 container closed"
>
  <div
    className="jsx-966561052 jsx-369727501 search"
  >
    <input
      className="jsx-966561052 jsx-369727501"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  <JSXStyle
    css="div.container.jsx-966561052{padding:4rem;}div.container.open.jsx-966561052{padding:2rem;position:fixed;background-color:rgba(0,0,0,1);top:0;bottom:0;left:0;right:0;zindex:12;margintop:3rem;}div.search.jsx-966561052{text-align:center;}input.jsx-966561052{background-color:rgba(0,0,0,0);border-style:none;width:80%;border-bottom:2px solid #fff;color:rgba(255,255,255,1);}input.jsx-966561052:focus{outline:none;}"
    styleId="966561052"
  />
  <JSXStyle
    css="input.__jsx-style-dynamic-selector{text-align:center;font-size:2rem;}"
    dynamic={
      Array [
        "center",
        "2rem",
      ]
    }
    styleId="3662217438"
  />
</div>
`;
